### PR TITLE
fix application reconcile

### PR DESCRIPTION
**Pull request by zackzhangkai**
Signed-off-by: zackzhang <zackzhang@yunify.com>

**What type of PR is this?**

1. When add component, it will not do reconcile if the component has no OwnerRef section. Use the `EnqueueRequestsFromMapFunc` instead of `EnqueueRequestForOwner`.

2. Application of kubernetes-sigs, is defined only by `app.kubernetes.io/name` `app.kubernetes.io/version` labels .  So the resources , such as ingress which has no `app` label, can be added to application.

3. Refactor utils of controller/virturalservices/utils to controller/utils. So the common codes can be used by other package, such as `application controller` and `destinationrule controller`.

/kind bug
/area microservice
/assign @zryfish

### CODE REVIEW

File: pkg/controller/virtualservice/util/util.go [hunk 1]
```diff
@@ -36,7 +36,6 @@ const (
 var ApplicationLabels = [...]string{
 	ApplicationNameLabel,
 	ApplicationVersionLabel,
-	AppLabel,
```
Comment (zryfish) [inline pkg/controller/virtualservice/util/util.go:39]: `app` label is required by istio, why do we remove it?

File: pkg/controller/application/application_controller.go [hunk 2]
```diff
@@ -133,9 +138,11 @@ func (r *ReconcileApplication) Reconcile(request reconcile.Request) (reconcile.R
 	return reconcile.Result{}, nil
 }
 
-func isApp(o metav1.Object) bool {
-	if o.GetLabels() == nil || !util.IsApplicationComponent(o.GetLabels()) {
-		return false
+func isApp(obs ...metav1.Object) bool {
+	for _, o := range obs {
+		if o.GetLabels() != nil && util.IsApplicationComponent(o.GetLabels()) {
```
Comment (zryfish) [inline pkg/controller/application/application_controller.go:143]: So, if any one of obs has required labels, return true, not all of them?

Comment (zackzhangkai) [reply pkg/controller/application/application_controller.go:143]: Yes.

File: pkg/controller/virtualservice/util/util.go (hunk 1 above)
Comment (zackzhangkai) [reply pkg/controller/virtualservice/util/util.go:39]: Use another AppLabels , which only have the necessary labels for kubernetes-sigs/application. So the resources ,such as ingress which has no `app` labels can be treated as application.

File: pkg/controller/destinationrule/destinationrule_controller.go [hunk 3]
```diff
@@ -19,6 +19,7 @@ package destinationrule
 import (
 	"context"
 	"fmt"
+	"kubesphere.io/kubesphere/pkg/models/servicemesh"
```
Comment (zryfish) [inline pkg/controller/destinationrule/destinationrule_controller.go:22]: why do we move util package into models? That would makes the controller depend on the models package.

Comment (zackzhangkai) [reply pkg/controller/destinationrule/destinationrule_controller.go:22]: There are some other controllers depend on the package `util`, such as application controller、destinationrule controller. The `util` package located under the folder of virtualservice controller . Its path is `pkg/controller/virtualservice/util/util.go`.  I think it's unresonable.

Comment (zackzhangkai) [reply pkg/controller/destinationrule/destinationrule_controller.go]: change `controller/virtualservice/util` to `controller/util/servicemesh`.

**Pull request merged**